scripts/apps/relations/services/RelationsService.ts

```diff
@@ -1,26 +1,29 @@
 import {pickBy} from 'lodash';
 import {IArticle} from 'superdesk-interfaces/Article';
 
-RelationsService.$inject = ['archiveService', 'mediaIdGenerator'];
+RelationsService.$inject = ['archiveService', 'mediaIdGenerator', 'api'];
 
-export function RelationsService(archiveService, mediaIdGenerator) {
+export function RelationsService(archiveService, mediaIdGenerator, api) {
     this.getRelatedItemsWithoutMediaGallery = function(item: IArticle, fields) {
         if (!item.associations) {
             return [];
         }
 
-        const relatedWithoutMedia = pickBy(item.associations, (value, key) => {
-            var parts = mediaIdGenerator.getFieldParts(key);
-            var field = fields.find((f) => f._id === parts[0]);
+        return api.find('archive', item._id)
```

Comment:
Why are you fetching the same item instead of associated items?

Reply:
Fetching one by one will take more time. Also, I am returning the item with the complete JSON of related items together.

Reply:
> Also, I am returning the item with the complete JSON of related items together.
It's a requirement that an item can only hold only the IDs and not the entire payloads, isn't it?

Reply:
In the database storing the ID only. Whenever we fetch the item then, we are retrieving the related item data based on that ID and then sending it to the frontend.

Reply:
This data structure should not be limited to the database. It should be the same in the API too.

I've tried checking out how it works, and it looks like it [saves the entire snapshots of related items even in the database](https://user-images.githubusercontent.com/2076953/51734237-3ba5bc00-2084-11e9-908e-b0ac2542eb70.gif). I've tested with commit `3634f472` from https://github.com/superdesk/superdesk-core/pull/1446 and `81ea8e91` from this PR.

Let me know if you can reproduce this.

Reply:
I can reproduce the same with server-core `3634f47`

Reply:
You are testing with `media-gallery` not with `related item`, I guess.

Reply:
You are right. I've tested again with text items. I can no longer see the payload, but there's a field `_updated` which should not be there since it's duplicating the one on the item itself.

And as I mentioned in the previous comment, only the id should appear in the API response instead of the entire payload which is returned right now.

Reply:
I also noticed that when you update an item, the entire payload of a related item is sent in the POST request when only an id should be included.